Sources_v3/StreamChat/Models/ExtraData.swift

```diff
@@ -11,8 +11,14 @@ public protocol ExtraData: Codable & Hashable {
 }
 
 /// A type representing no extra data for the given model object.
-public struct NoExtraData: Codable, Hashable, UserExtraData, ChannelExtraData, MessageExtraData, MessageReactionExtraData,
-    AttachmentExtraData {
+public struct NoExtraData: Codable,
+                           Hashable,
+                           UserExtraData,
+                           ChannelExtraData,
+                           MessageExtraData,
+                           MessageReactionExtraData,
+                           AttachmentExtraData,
+                           ExtraDataTypes {
```

Comment:
Implementing `ExtraDataTypes` by the type creates a recursive dependency when nested types can be accessed from any nested one since all of them are `NoExtraData` (e.g. `NoExtraData.User.Message.Channel`). Are we OK with that? (looks like something we don't want IMHO)

One way to avoid this is having concrete separate `XXXNoExtraData` types for each model which will implement just the `XXXExtraDataType`.

Reply:
I think we can live with this for now 👍